pkg/gbpserver/watchers/netflow_watch.go

```diff
@@ -0,0 +1,205 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchers
+
+import (
+	"fmt"
+	"github.com/noironetworks/aci-containers/pkg/gbpserver"
+	netflowpolicy "github.com/noironetworks/aci-containers/pkg/netflowpolicy/apis/aci.netflow/v1alpha"
+	netflowclientset "github.com/noironetworks/aci-containers/pkg/netflowpolicy/clientset/versioned"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	netflowCRDSub       = "NetflowExporterConfig"
+	netflowCRDParentSub = "PlatformConfig"
+)
+
+type netflowCRD struct {
+	DstAddr           string `json:"destIp"`
+	DstPort           int    `json:"destPort"`
+	Version           string `json:"flowType,omitempty"`
+	ActiveFlowTimeOut int    `json:"activeFlowTimeOut,omitempty"`
+	IdleFlowTimeOut   int    `json:"idleFlowTimeOut,omitempty"`
+	SamplingRate      int    `json:"samplingRate,omitempty"`
+	Name              string `json:"name,omitempty"`
+}
+
+type NetflowWatcher struct {
+	log *log.Entry
+	gs  *gbpserver.Server
+	idb *intentDB
+	rc  restclient.Interface
+}
+
+func NewNetflowWatcher(gs *gbpserver.Server) (*NetflowWatcher, error) {
+	gcfg := gs.Config()
+	level, err := log.ParseLevel(gcfg.WatchLogLevel)
+	if err != nil {
+		panic(err.Error())
+	}
+	logger := log.New()
+	logger.Level = level
+	log := logger.WithField("mod", "NETFLOW-W")
+	cfg, err := restclient.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	netflowclient, err := netflowclientset.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	restClient := netflowclient.AciV1alpha().RESTClient()
+	return &NetflowWatcher{
+		log: log,
+		rc:  restClient,
+		gs:  gs,
+		idb: newIntentDB(gs, log),
+	}, nil
+}
+
+func (nfw *NetflowWatcher) InitNetflowInformer(stopCh <-chan struct{}) error {
+	nfw.watchNetflow(stopCh)
+	return nil
+}
+
+func (nfw *NetflowWatcher) watchNetflow(stopCh <-chan struct{}) {
+
+	NetflowLw := cache.NewListWatchFromClient(nfw.rc, "netflowpolicies", metav1.NamespaceAll, fields.Everything())
+	_, netflowInformer := cache.NewInformer(NetflowLw, &netflowpolicy.NetflowPolicy{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				nfw.log.Infof("netflow added")
+				nfw.netflowAdded(obj)
+			},
+			UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+				nfw.log.Infof("netflow updated")
+				nfw.netflowAdded(newobj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				nfw.log.Infof("netflow deleted")
+				nfw.netflowDeleted(obj)
+			},
+		})
+	go netflowInformer.Run(stopCh)
+}
+
+func (nfw *NetflowWatcher) netflowAdded(obj interface{}) {
+	netflow, ok := obj.(*netflowpolicy.NetflowPolicy)
+	if !ok {
+		nfw.log.Errorf("netflowAdded: Bad object type")
+		return
+	}
+
+	nfw.log.Infof("netflowAdded - %s", netflow.ObjectMeta.Name)
```

Comment:
Can you add the namespace to these logs as well

Reply:
Since netflow is cluster scoped, namespace metadata is not stored and hence cant be used to log